driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -404,6 +397,7 @@ void cancel() {
 
         @Override
         public Message.Request request() {
+            Message.Request request = callback.request();
```

Comment:
Yes, but not here, rather in `Connection.write()`. Here we are in a higher level IMO, so we should only copy the original request if something in it should be modified, e.g. the CL.